docs/operations/azure_agents_en.md

```diff
@@ -0,0 +1,315 @@
+domainCredentialsMap = SystemCredentialsProvider.getInstance().getDomainCredentialsMap();
+domainCredentials = domainCredentialsMap.get(Domain.global());
+
+jenkinsAgentAdminCredentials = new UsernamePasswordCredentialsImpl(
+        CredentialsScope.GLOBAL,
+        jenkinsAgentAdminCredentialsID,
+        jenkinsAgentAdminCredentialsDescription,
+        jenkinsAgentAdminCredentialsUsername,
+        jenkinsAgentAdminCredentialsPassword
+)
+
+AzureCredentials azureCredentials = new AzureCredentials(
+        CredentialsScope.GLOBAL,
+        azureCredentialsID,
+        azureCredentialsDescription,
+        azureCredentialsSubscriptionID,
+        azureCredentialsClientID,
+        Secret.fromString(azureCredentialsClientSecret));
+azureCredentials.setTenant(azureCredentialsTenantID);
+
+systemCredentialStore.addCredentials(domain, jenkinsAgentAdminCredentials)
+domainCredentials.add(azureCredentials);
+SystemCredentialsProvider.getInstance().setDomainCredentialsMap(domainCredentialsMap);
+
+
+
+// Configure Azure Cloud
+// See https://plugins.jenkins.io/azure-vm-agents/#plugin-content-configure-vm-template-using-groovy-script
+// See https://github.com/jenkinsci/azure-vm-agents-plugin/blob/master/src/main/java/com/microsoft/azure/vmagent/builders/AzureVMCloudBuilder.java
+def azureTemplate = new AzureVMTemplateBuilder()
+        .withName("jenkins-agent-azure")
+        .withLabels("docker node linux ubuntu")
+        .withLocation("West Europe")
+        .withVirtualMachineSize("Standard_DS3_v2")
+        .withExistingStorageAccount("jenkinsagentstor")
+        .addNewAdvancedImage()
+        .withCustomManagedImage("/subscriptions/${azureCredentialsSubscriptionID}/resourceGroups/${azureResourceGroup}/providers/Microsoft.Compute/images/cesJenkinsDockerAgent")
+        .withNetworkSecurityGroupName("Jenkins-Agents_NSG")
+        .endAdvancedImage()
+        .withAdminCredential(jenkinsAgentAdminCredentialsID)
+        .build()
+def azureCloud = new AzureVMCloudBuilder()
+        .withCloudName("Azure Cloud")
+        .withAzureCredentialsId(azureCredentialsID)
+        .withExistingResourceGroupName(azureResourceGroup)
+        .addToTemplates(azureTemplate)
+        .build()
+Jenkins.getInstance().clouds.add(azureCloud)
+
+```
+
+- Insert the Azure credentials from the steps before
+- Make sure the resource group name is correct
+- Run the script via the Jenkins script console or via the additional scripts directory as described above
+
+## Testing with a Jenkins pipeline
+
+You can test if everything worked out correctly by using this test-pipeline:
+- Add a Jenkins Pipeline project
+- Use the following code for the pipeline:
+
+```groovy
+#!groovy
+@Library(['github.com/cloudogu/ces-build-lib@1.62.0'])
+
+import com.cloudogu.ces.cesbuildlib.*
+
+node('docker') {
+    stage('Clone Repository') {
+        checkout scmGit(userRemoteConfigs: [
+            [url: 'https://github.com/cloudogu/sonar']
```

Comment:
This repository may not be suitable for an example because the building needs instance credentials to be configured. Maybe we can find a different repository with a Dockerfile which is easier to build.